cmd/kube-proxy/app/server.go

```diff
@@ -350,10 +369,12 @@ with the apiserver API to configure the proxy.`,
 			if err := initForOS(opts.WindowsService); err != nil {
 				glog.Fatalf("failed OS init: %v", err)
 			}
-
 			if err := opts.Complete(); err != nil {
 				glog.Fatalf("failed complete: %v", err)
 			}
+			if err := opts.ProcessArgs(); err != nil {
```

Comment:
Instead of adding a new top-level function to the complete/validate/run flow, I'd recommend renaming `ProcessArgs()` to `processHostnameOverrideFlag()` and calling it from within `Complete()`